airbyte-integrations/connectors/source-bigquery/source_bigquery/streams.py

```diff
@@ -327,95 +557,165 @@ def _updated_state(self, current_stream_state: MutableMapping[str, Any], latest_
         """
         if not self.cursor_field:
             return self.state
-        
         latest_record_state = latest_record[self.cursor_field]
         stream_state = current_stream_state.get(self.cursor_field, None)
-
         if stream_state:
             self._cursor = max(latest_record_state, stream_state)
         else:
             self._cursor = latest_record_state
-        self.state = {self.cursor_field: self._cursor} 
-
+        self.state = {self.cursor_field: self._cursor}
         if datetime.now() >= self._checkpoint_time + timedelta(minutes=15):
             self.checkpoint(self.name, self.state, self.namespace)
             self._checkpoint_time = datetime.now()
-
         return self.state
 
+    def _extract_borders(self):
+        if not self.start_time and not self.end_time:
+            first_record = TableQueryRecord(self.project_id, self.stream_name, "ASC", self.cursor_field, self.where_clause, authenticator=self._auth)
+            last_record = TableQueryRecord(self.project_id, self.stream_name, "DESC", self.cursor_field, self.where_clause, authenticator=self._auth)
+            self.start_time = next(first_record.read_records(sync_mode=SyncMode.full_refresh)).get(self.cursor_field, None)
```

Comment:
It seems odd that we perform HTTP request to get those. Isn't it the state supposed to describe this information? 

It also seems like the state I input to the connector is not taken into account. Is this expected?

Reply:
> which will not give me the timestamp of the last record

Why do we need the timestamp if we have the most recent cursor field value?

Reply:
I will make an update so that we do not send an http request if we have a state set. For the end timestamp I will use the current time as you suggested in our last meeting